components/keys/src/types.rs

```diff
@@ -211,14 +212,108 @@ impl Display for Key {
     }
 }
 
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct TimeStamp(u64);
+
+const TSO_PHYSICAL_SHIFT_BITS: u64 = 18;
```

Comment:
I don't think it is a good idea to expose so many details of ts in TiKV.

Reply:
If some day PD change the rule or change the meaning of ts, TiKV should keep updated, but it is easy to forget.

Reply:
This constant existed before it is not new, it has only been moved. Also it is not public.

Reply:
Our check_txn_status and txn_heart_beat API needs to determine whether a ts has its TTL expired, so TiKV needs to know how ts represents physical time. This thing is added to TiKV when we are implementing that two APIs.